### PR TITLE
bears/yaml: Add TravisLintBear

### PR DESCRIPTION
**Pull request by yash-nisar**
Closes https://github.com/coala/coala-bears/issues/294

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: bears/yaml/TravisLintBear.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+from coalib.bearlib.abstractions.Linter import linter
+from dependency_management.requirements.GemRequirement import GemRequirement
+
+
+@linter(executable='travis',
+        output_format='regex',
+        output_regex=r'\[x\]\s+(?P<message>.+)')
```
Comment (Makman2) [inline bears/yaml/TravisLintBear.py]: that's all information travis-lint provides? :/

File: bears/yaml/TravisLintBear.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+class TravisLintBear:
+    """
+    A validator for your .travis.yml that attempts to reduce common build
```
Comment (Makman2) [inline bears/yaml/TravisLintBear.py]: maybe enclose `.travis.yml` in double backticks?

File: tests/yaml/TravisLintBearTest.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
+from queue import Queue
+
+from bears.yaml.TravisLintBear import TravisLintBear
+from coalib.results.Result import Result
+from coalib.results.RESULT_SEVERITY import RESULT_SEVERITY
+from coalib.settings.Section import Section
+from coalib.testing.LocalBearTestHelper import LocalBearTestHelper
+from coalib.testing.BearTestHelper import generate_skip_decorator
+from coala_utils.ContextManagers import prepare_file
+
```
Comment (Makman2) [inline tests/yaml/TravisLintBearTest.py]: another newline plase ;) 2 newlines between imports and code :+1:

File: tests/yaml/TravisLintBearTest.py [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+deploy:
+  provider: rubygems
+  api_key:
+    secure: abcdef
+  gem: travis
+  on:
+    repo: xyz/abc
+    ruby: 2.0.0
+    condition: "$(uname) = Linux"
+"""
```
Comment (Makman2) [inline tests/yaml/TravisLintBearTest.py]: I think a separate file would be good.

File: tests/yaml/TravisLintBearTest.py [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
```
Comment (Makman2) [inline tests/yaml/TravisLintBearTest.py]: same here --> separate file

File: bears/yaml/TravisLintBear.py (hunk 1 above)
Comment (yash-nisar) [inline bears/yaml/TravisLintBear.py]: Yes, even I did not like it at first, but there is no better alternative. See https://github.com/coala/coala-bears/issues/294#issuecomment-304322328

File: bears/yaml/TravisLintBear.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
+from coalib.bearlib.abstractions.Linter import linter
```
Comment (Makman2) [inline bears/yaml/TravisLintBear.py]: ah newline please here, as `dependency_management` is a new package :+1:

**Comment by yash-nisar** [comment]
Updated @Makman2

**Comment by yash-nisar** [comment]
Damnit gitmate.

File: tests/yaml/travislint_test_files/.good_travis.yml [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+- 1.9.3
+- 2.0.0
+- 2.1.0
+- 2.1.3
+- jruby
+os:
+- linux
+- osx
+matrix:
+  exclude:
```
Comment (meetmangukiya) [inline tests/yaml/travislint_test_files/.good_travis.yml]: is YAMLLint insane?

Comment (yash-nisar) [inline tests/yaml/travislint_test_files/.good_travis.yml]: Haha drunk rather.

File: bears/yaml/TravisLintBear.py [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+        output_format='regex',
+        output_regex=r'\[x\]\s+(?P<message>.+)')
+class TravisLintBear:
+    """
+    A validator for your ``.travis.yml`` that attempts to reduce common build
+    errors such as :
+
+    - invalid YAML
+    - missing language key
+    - unsupported runtime versions of Ruby, PHP, OTP, etc
```
Comment (Asnelchristian) [inline bears/yaml/TravisLintBear.py]: `etc.`

File: tests/yaml/TravisLintBearTest.py [hunk 9]
```diff
@@ -0,0 +1,72 @@
+import os
+from queue import Queue
+
+from bears.yaml.TravisLintBear import TravisLintBear
```
Comment (Makman2) [inline tests/yaml/TravisLintBearTest.py]: own package import ;)
below coalib imports with a newline in between ;)

**Comment by Asnelchristian** [comment]
the KeywordBear is messing

File: bears/yaml/TravisLintBear.py [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+from dependency_management.requirements.GemRequirement import GemRequirement
+
+
+@linter(executable='travis',
+        output_format='regex',
+        output_regex=r'\[x\]\s+(?P<message>.+)')
+class TravisLintBear:
+    """
+    A validator for your ``.travis.yml`` that attempts to reduce common build
+    errors such as :
```
Comment (Makman2) [inline bears/yaml/TravisLintBear.py]: obsolete space between `as` and colon

**Comment by yash-nisar** [comment]
Updated @Makman2

**Comment by Makman2** [comment]
ack 2f004bc

**Comment by Makman2** [comment]
@rultor merge

**Pull request merged**